client/src/contexts/UserContext.js

```diff
@@ -1,13 +1,103 @@
-import React, { createContext, useState, useMemo } from "react";
+import React, { createContext, useState, useMemo, useEffect } from "react";
+import { getBoardShallow, getBoard } from "../API/board";
+import { getUser } from "../API/user";
 
 export const UserContext = createContext({});
 
 export const UserContextProvider = (props) => {
   const [loggedIn, setLoggedIn] = useState(false);
   const [user, setUser] = useState({});
+  const [userId, setUserId] = useState("");
+  const [boardList, setBoardList] = useState([]);
+  const [currBoardId, setCurrBoardId] = useState("");
+  const [currBoard, setCurrBoard] = useState(null);
+  const [createCount, setCreateCount] = useState(0);
+
+  useEffect(() => {
+    if (userId && createCount > 0) {
+      async function getData() {
+        const res = await getUser(userId);
+        const updatedUser = res.data;
+        setUser(updatedUser);
+      }
+      getData();
+    }
+  }, [userId, createCount]);
+
+  useEffect(() => {
+    const getAllBoards = async () => {
+      const allBoards = {};
+      if (user && user.boards) {
+        setUserId(user._id);
+        const pending = user.boards.map(async (boardId) => {
+          return await getBoardShallow(boardId);
+        });
+        const resolved = await Promise.all(pending);
+        resolved.map((res) => (allBoards[res.data._id] = res.data));
+        const boardList = user.boards.map((boardId) => {
+          if (allBoards[boardId]) {
+            return allBoards[boardId];
+          }
+          return null;
+        });
+        setBoardList(boardList);
+        setCurrBoardId(currBoardId || boardList[0]._id);
+      }
+    };
+    getAllBoards();
+  }, [user, currBoardId]);
```

Comment:
do we want to be fetching all of a user's boards every time currBoardId changes?

Reply:
no we do not 👍